### PR TITLE
crimson/admin/admin_socket: remove path file if it exists

### PR DESCRIPTION
**Pull request by Matan-B**
Fixes: https://tracker.ceph.com/issues/62525

https://pulpito.ceph.com/matan-2023-08-22_16:08:32-crimson-rados-wip-matanb-crimson-asock-cleanup-distro-crimson-smithi/

Before this patch:
```
DEBUG 2023-08-21 10:37:04,849 [shard 0] osd - start: asok socket path=/var/run/ceph/ceph-osd.1.asok
ERROR 2023-08-21 10:37:04,849 [shard 0] osd - start: unable to listen(/var/run/ceph/ceph-osd.1.asok): posix_listen failed for address /var/run/ceph/ceph-osd.1.asok: Address already in use
```

After:
```
DEBUG 2023-08-22 17:03:45,213 [shard 0] osd - operator(): Admin Socket socket path=/var/run/ceph/ceph-osd.1.asok already exists, removing
DEBUG 2023-08-22 17:03:45,213 [shard 0] osd - operator(): asok socket path=/var/run/ceph/ceph-osd.1.asok
```

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/admin/admin_socket.cc [hunk 1]
```diff
@@ -231,41 +231,50 @@ seastar::future<> AdminSocket::start(const std::string& path)
...
-        connected_sock = std::move(acc.connection);
-        return seastar::do_with(connected_sock->input(),
-                                connected_sock->output(),
-          [this](auto& input, auto& output) mutable {
-          return handle_client(input, output);
-        }).finally([this] {
-          assert(connected_sock.has_value());
-          connected_sock.reset();
+  return seastar::file_exists(path).then([this, path](auto exists) {
+    auto fut = seastar::now();
```
Comment (ronen-fr) [inline src/crimson/admin/admin_socket.cc:235]: now() here  vs make_ready_future() a few lines below

**Comment by ronen-fr** [comment]
LGTM (and 2 nits)

**Review by rzarzynski — APPROVED** [review]

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2023-08-24_09:25:19-crimson-rados-wip-matanb-crimson-testing-24.8-v2-distro-crimson-smithi/

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
Merging based on:
https://pulpito.ceph.com/matan-2023-08-24_14:53:35-crimson-rados-wip-matanb-crimson-testing-24.8-v3-distro-crimson-smithi/

Failed, unrelated:
7378557, 7378563, 7378597, 7378614 - https://tracker.ceph.com/issues/61651
7378571, 7378575, 7378581, 7378611  - https://tracker.ceph.com/issues/62594
7378560, 7378603, 7378586, 7378601, 7378607, 7378611 - https://tracker.ceph.com/issues/62595

**Pull request merged**